### PR TITLE
Add the Gemma 2 model

### PR DESCRIPTION
**Pull request by EricLBuehler**

### CODE REVIEW

File: mistralrs-core/src/pipeline/normal_loaders.rs [hunk 1]
```diff
@@ -741,3 +745,52 @@ impl NormalModelLoader for Qwen2Loader {
...
+
+impl NormalModelLoader for Gemma2Loader {
+    fn load(
+        &self,
+        config: &str,
+        use_flash_attn: bool,
+        vb: VarBuilder,
+        normal_loading_metadata: NormalLoadingMetadata,
+    ) -> Result<Box<dyn NormalModel + Send + Sync>> {
+        if use_flash_attn {
```
Comment (michaelfeil) [inline mistralrs-core/src/pipeline/normal_loaders.rs:764]: @EricLBuehler I added softcap FA2: https://github.com/huggingface/candle/tree/main/candle-flash-attn/kernels

**Review by michaelfeil — COMMENTED** [review]
FYI

File: mistralrs-core/src/pipeline/normal_loaders.rs (hunk 1 above)
Comment (EricLBuehler) [reply mistralrs-core/src/pipeline/normal_loaders.rs:764]: @michaelfeil thanks! I actually added softcap support to the Candle fork we use already: 

https://github.com/EricLBuehler/mistral.rs/blob/master/mistralrs-core%2Fsrc%2Fattention.rs#L56

I saw your 3 PRs to Candle updating the FA kernels though, and noticed you mentioned you were working on FA3 [here](https://github.com/michaelfeil/candle-flash-attn-v3)! I'd be super excited to integrate that here.

Comment (michaelfeil) [reply mistralrs-core/src/pipeline/normal_loaders.rs:764]: Hey @EricLBuehler, FA-3 is working! Just added a permissive license, enjoy!

**Pull request merged**